### PR TITLE
[WIP]Benchmark: adding ipfs extract benchmark using unixFS engine

### PR DESCRIPTION
**Pull request by litzenberger**
Benchmark: adding ipfs extract benchmark using unixFS engine

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env node
+'use strict'
+
+const localExtract = require('./tests/local-extract')
+const os = require('os')
+
+// Run through use cases
+const logger = require('pino')()
```
Comment (mcollina) [inline index.js]: why is pino needed?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const results = []
+async function benchmark() {
+
+  results.push(await localExtract(logger, "unixFS:extract:smallfile", "tests/fixtures/200Bytes.txt"))
```
Comment (mcollina) [inline index.js]: how long does this take? I think we should do a series of run, mainly to ensure some relevance of the results.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  results[0].cpu = os.cpus(),
+    results[0].loadAvg = os.loadavg()
+
+  results.push(await localExtract(logger, "unixFS:extract:largefile", "tests/fixtures/1.2MiB.txt"))
+  results[1].cpu = os.cpus(),
+    results[1].loadAvg = os.loadavg()
+  console.log(results)
```
Comment (mcollina) [inline index.js]: I think we should be producing JSON data.

File: lib/create-node.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  const peer = new IPFS({
+    repo: '/tmp/peer',
+    config: config,
+    init: {
+      emptyRepo: true
+    }
+  })
+  peer.on('ready', () => {
+    resolve(peer)
+  })
```
Comment (mcollina) [inline lib/create-node.js]: what happens if it can’t connect? I think we should be rejecting.

File: lib/create-node.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env node
+'use strict'
+// TODO: point this to the branch code
+const IPFS = require('../../js-ipfs')
+const config = require('../config/default-config.json')
+
+module.exports = new Promise((resolve) => {
```
Comment (mcollina) [inline lib/create-node.js]: can you transform this into a factory?

File: lib/create-node.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env node
+'use strict'
+// TODO: point this to the branch code
+const IPFS = require('../../js-ipfs')
+const config = require('../config/default-config.json')
```
Comment (mcollina) [inline lib/create-node.js]: I prefer if the config comes as parameters and it’s not required in.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "ipfs-benchmarks",
+  "version": "0.1.0",
+  "description": "Benchmarks for js-ifps",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (mcollina) [inline package.json]: can you make this run standard?

File: tests/local-extract.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const fs = require('fs')
+const prettyHrtime = require('pretty-hrtime')
+const node = require('../lib/create-node.js')
+
+
+const ora = require('ora')
+const handler = "local extract"
+
+function localExtract(logger, name, file) {
+  return new Promise(async (resolve) => {
```
Comment (mcollina) [inline tests/local-extract.js]: there is no need for this promise, make localExtract an async function

File: tests/local-extract.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    const spinner = ora(`Started ${handler}`).start()
+    spinner.color = 'magenta'
+    spinner.text = "testing"
```
Comment (mcollina) [inline tests/local-extract.js]: can you move the spinner into the runner into index.js?

File: index.js (hunk 3 above)
Comment (litzenberger) [reply index.js]: done

File: lib/create-node.js (hunk 6 above)
Comment (litzenberger) [reply lib/create-node.js]: I can do it as parameters but this will get very big once we have multi peers setup.

File: tests/local-extract.js (hunk 9 above)
Comment (litzenberger) [reply tests/local-extract.js]: yes, still in refactoring stage.

File: index.js (hunk 1 above)
Comment (Elexy) [reply index.js]: I see that you put an index.js in the root. My idea was to have the tests al reside under `/tests`, the runner under `/runner` and use ansible to sync /tests to VMs (benchmarks/clinic) and the run them remotely using ssh. What do you think?

File: tests/local-extract.js (hunk 9 above)
Comment (Elexy) [reply tests/local-extract.js]: possibly put it behind a `verbose` param, run interactively it looks nice but CI is not gonna care to see it.

**Comment by litzenberger** [comment]
@Elexy what is the advantage to running each test over ssh vs just running it from inside the docker container?

File: index.js (hunk 2 above)
Comment (litzenberger) [reply index.js]: we are going to use a different runner

File: tests/package.json [hunk 10]
```diff
@@ -1,16 +1,23 @@
...
-  "name": "local-transfer",
-  "version": "1.0.0",
-  "description": "",
+  "name": "ipfs-benchmarks",
+  "version": "0.1.0",
+  "description": "Benchmarks for js-ifps",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (mcollina) [inline tests/package.json]: can you ran standard here? Also, the application would need some linting.

File: tests/local-add.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    )
+  }
+  catch (err) {
+    throw Error(err)
+  }
+
+}
+const results = []
+
+// mock scenerios until the other one is working.
```
Comment (mcollina) [inline tests/local-add.js]: why? can you please explain?

File: tests/local-extract.js [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    throw Error(err)
+  }
+
+}
+const results = []
+
+// mock scenerios until the other one is working.
+async function scenarios() {
+
+  const spinner = ora(`Started `).start()
```
Comment (mcollina) [inline tests/local-extract.js]: I think the spinner and logic for printing the data could be factored out.

File: tests/results10.json [hunk 13]
```diff
@@ -0,0 +1,410 @@
+[
+  {
```
Comment (mcollina) [inline tests/results10.json]: Are these files needed?

File: lib/create-node.js (hunk 5 above)
Comment (litzenberger) [reply lib/create-node.js]: Merging this but will do it on next branch
